<a href="https://colab.research.google.com/github/djbell01/563-DawsonBell/blob/main/Problem_Set_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import numpy as np

$1.1$

Consider a system $A$ consisting of subsystems $A_{1}$ and $A_{2}$, for which $\Omega_{1} = 10^{20}$ and $\Omega_{2} = 10^{22}$. What is the number of configurations available to the combined system? Also, compute the entropies $S$, $S_{1}$, and $S_{2}$.


$$Ω_{combined} = Ω_{1} \cdot Ω_{2}$$

$$S = k_{B}\ln(Ω)$$

In [ ]:
omega_1 = 10e20
omega_2 = 10e22
k_B = 1.381e-23 # units of J/K

combined_states = omega_1*omega_2

S_1 = k_B*np.log(omega_1)
S_2 = k_B*np.log(omega_2)
S = k_B*np.log(combined_states)

print(f'Combined number of states = {combined_states}')
print(f'Entropy of A_1 = {S_1} J/K')
print(f'Entropy of A_2 = {S_2} J/K')
print(f'Combined system entropy = {S} J/K')

Combined number of states = 9.999999999999999e+43
Entropy of A_1 = 6.677727028192031e-22 J/K
Entropy of A_2 = 7.3137010308769875e-22 J/K
Combined system entropy = 1.3991428059069019e-21 J/K


$1.2$

By what factor does the number of available configurations increase when $10\ m^{3}$ of air at $1.0\ atm$ and $300\ K$ is allowed to expand by $0.001\%$ at constant temperature?

$$S = Nk_{B}\ln\biggl(\frac{Ve^{5/2}}{Nλ_{th}^{3}}\biggr)$$

$$S = k_{B}\ln(Ω)$$

$$k_{B}ln(Ω) = Nk_{B}\ln\biggl(\frac{Ve^{5/2}}{Nλ_{th}^{3}}\biggr)$$

$$e^{ln(Ω)} = e^{N\ln(\frac{Ve^{5/2}}{Nλ_{th}^{3}})}$$

$$Ω = e^{\ln((\frac{Ve^{5/2}}{Nλ_{th}^{3}})^{N})}$$

$$Ω = \biggl(\frac{Ve^{5/2}}{Nλ_{th}^{3}}\biggl)^{N}$$

$$\frac{Ω_{2}}{Ω_{1}} = \frac{\biggl(\frac{V_{2}e^{5/2}}{Nλ_{th}^{3}}\biggl)^{N}}{\biggl(\frac{V_{1}e^{5/2}}{Nλ_{th}^{3}}\biggl)^{N}}$$

$$\frac{Ω_{2}}{Ω_{1}} = \Bigg(\frac{\frac{V_{2}e^{5/2}}{Nλ_{th}^{3}}}{\frac{V_{1}e^{5/2}}{Nλ_{th}^{3}}}\Bigg)^{N}$$

$$\frac{Ω_{2}}{Ω_{1}} = \bigg(\frac{V_{2}}{V_{1}}\bigg)^{N}$$

In [ ]:
P = 101325 # Pa
T = 300 # K
V_1 = 10 # m^3
k_B = 1.38065E-23 # J/K

V_2 = V_1*1.00001 # m^3
N = (P*V_1)/(k_B*T)


fraction = (V_2/V_1)
print(f'Factor: {fraction}^{N}')

Factor: 1.00001^2.446311519936262e+26


$1.3$

By what factor does the number of available configurations increase when $150\ kJ$ is added to a system containing $2.0\ mol$ of particles at constant volume and $T = 300\ K$?

<center> Given the following four functions... </center>

$$S = k_{B}\lnΩ$$

$$S = \frac{U - F}{T}$$

$$U = \frac{3}{2}Nk_{B}T$$

$$F = -Nk_{B}T\ln\bigg(\frac{Ve}{Nλ_{th}^{3}}\bigg)$$

<center>...it can be shown that the number of states is independent of the energy </center>

$$k_{B}\lnΩ = \frac{U - F}{T}$$

$$\lnΩ = \frac{U - F}{k_{B}T}$$

$$Ω = e^{(\frac{U - F}{k_{B}T})}$$

$$\frac{Ω_{2}}{Ω_{1}} = \frac{e^{(\frac{U_{2} - F_{2}}{k_{B}T_{2}})}}{e^{(\frac{U_{1} - F_{1}}{k_{B}T_{1}})}}$$

$$\frac{Ω_{2}}{Ω_{1}} = e^{(\frac{U_{2} - F_{2}}{k_{B}T_{2}})}\cdot e^{(\frac{F_{1} - U_{1}}{k_{B}T_{1}})}$$

$$\frac{Ω_{2}}{Ω_{1}} = e^{\bigg(\frac{\frac{3}{2}Nk_{B}T_{2} + Nk_{B}T_{2}\ln(\frac{Ve}{Nλ_{th}^{3}})}{k_{B}T_{2}}\bigg)}\cdot e^{\bigg(\frac{-Nk_{B}T_{1}\ln(\frac{Ve}{Nλ_{th}^{3}}) - \frac{3}{2}Nk_{B}T_{1}}{k_{B}T_{1}}\bigg)}$$

$$\frac{Ω_{2}}{Ω_{1}} = e^{\bigg(\frac{3}{2}N + N\ln(\frac{Ve}{Nλ_{th}^{3}})\bigg)}\cdot e^{\bigg(-N\ln(\frac{Ve}{Nλ_{th}^{3}}) - \frac{3}{2}N\bigg)}$$

$$\frac{Ω_{2}}{Ω_{1}} = e^{0} = 1$$

In [ ]:
E_added = 150 #kJ
R = 0.008314 #kJ/molK
T_1 = 300 #K
mol = 2 #mol
avogadro = 6.022E23 # 1/mol

E_initial = R*T_1*mol
N = mol*avogadro

delta_T = E_added/(3/2*N*k_B)
T_2 = T_1 + delta_T





$1.3$ alternative

$$ΔS = \frac{ΔU}{T}$$

$$ΔS = k_{B}\ln\frac{Ω_{2}}{Ω_{1}}$$

$$k_{B}\ln\frac{Ω_{2}}{Ω_{1}} = \frac{ΔU}{T}$$

$$\frac{Ω_{2}}{Ω_{1}} = e^{\frac{ΔU}{k_{B}T}}$$

In [7]:
delta_U = 150000 # J
T = 300 # K
k_B = 1.38065E-23 # J/K

delta_S = delta_U/T
nat_log_factor = delta_S/k_B

print(f'Factor: e^{nat_log_factor}')

Factor: e^3.6214826349907653e+25


$1.4$

A sample consisting of five molecules has a total energy $5ϵ$. Each molecule is able to occupy states of energy $ϵj$, with $j = 0,1,2,...,∞$. Draw up a table with columns by the energy of the states and write beneath them all configurations that are consistent with the total energy. Identify the type of configuration that is most probable.